Lib/subprocess.py

```diff
@@ -1307,15 +1307,18 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
                 try:
                     exception_name, hex_errno, err_msg = (
                             errpipe_data.split(b':', 2))
+                    # The encoding here should match the encoding
+                    # written in by the subprocess implementations
+                    # like _posixsubprocess
+                    err_msg = err_msg.decode()
```

Comment:
Yeah, I'm not sure where you're expecting this junk from. If the split managed to succeed then the data is likely written in by `_posixsubprocess` here https://github.com/python/cpython/blob/master/Modules/_posixsubprocess.c#L522
If it is then there wouldn't be any extra junk and so the decode should never fail unless the junk somehow manages to contain 2 colon characters by accident.